src/main/java/manufacturing/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package manufacturing.dao;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import manufacturing.db.Storage;
+import manufacturing.injections.Dao;
+import manufacturing.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    public Manufacturer create(Manufacturer manufacturer) {
+        Long id = Storage.manufacturerId++;
+        manufacturer.setId(id);
+        Storage.getManufacturers().put(id, manufacturer);
+        return manufacturer;
+    }
+
+    public Optional<Manufacturer> get(Long id) {
+        return Optional.ofNullable(Storage.getManufacturers().get(id));
+    }
+
+    public List<Manufacturer> getAll() {
+        return new ArrayList<>(Storage.getManufacturers().values());
+    }
+
+    public Manufacturer update(Manufacturer newManufacturer) {
+        Manufacturer manufacturerToChange =
+                Storage.getManufacturers().get(newManufacturer.getId());
+        if (manufacturerToChange == null) {
+            throw new RuntimeException("Manufacturer " + newManufacturer.getId() + " not found");
+        }
+        manufacturerToChange.setCountry(newManufacturer.getCountry());
```

Comment:
better replace newManufacturer in the map:
```
Storage.getManufacturers().put(newManufacturer.getId(), newManufacturer);
```
this code is more flexible. We shouldn't change the logic of this method if the manufacturer object is changed